test/e2e/postinstall/syncsets/syncsets_suite_test.go

```diff
@@ -0,0 +1,517 @@
+				err = hiveClient.Delete(ctx, syncSetWithUpsertApplyMode)
+				Ω(err).ShouldNot(HaveOccurred())
+
+				time.Sleep(5 * time.Second)
+
+				By("Verifying the ConfigMap was not deleted on the target cluster")
+				resultConfigMap = &corev1.ConfigMap{}
+				err = targetClusterClient.Get(ctx, client.ObjectKey{Name: "foo", Namespace: "default"}, resultConfigMap)
+				Ω(err).ShouldNot(HaveOccurred())
+				Ω(resultConfigMap.Data).Should(Equal(map[string]string{"foo": "bar"}))
+
+			})
+
+			It("Test SyncSet patches", func() {
+				ctx := context.TODO()
+				configMap := &corev1.ConfigMap{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "ConfigMap",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: "default",
+					},
+					Data: map[string]string{
+						"foo": "bar",
+					},
+				}
+				SyncSetPatch := &hivev1.SyncSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-syncpatch",
+						Namespace: clusterNamespace,
+					},
+					Spec: hivev1.SyncSetSpec{
+						ClusterDeploymentRefs: []corev1.LocalObjectReference{
+							{
+								Name: clusterName,
+							},
+						},
+						SyncSetCommonSpec: hivev1.SyncSetCommonSpec{
+							ResourceApplyMode: "Sync",
+							Patches: []hivev1.SyncObjectPatch{
+								{
+									APIVersion: "v1",
+									Kind:       "ConfigMap",
+									Name:       "foo",
+									Namespace:  "default",
+									Patch:      `{ "data": { "foo": "new-bar" } }`,
+									PatchType:  "merge",
+								},
+							},
+						},
+					},
+				}
+				By("Creating a resource ConfigMap on target cluster")
+				err := targetClusterClient.Create(ctx, configMap)
+				Ω(err).ShouldNot(HaveOccurred())
+				time.Sleep(5 * time.Second)
+
+				By("Verify resource ConfigMap created on target cluster")
+				resultConfigMap := &corev1.ConfigMap{}
+				err = targetClusterClient.Get(ctx, client.ObjectKey{Name: "foo", Namespace: "default"}, resultConfigMap)
+				Ω(err).ShouldNot(HaveOccurred())
+				Ω(resultConfigMap.Data).Should(Equal(map[string]string{"foo": "bar"}))
+
+				By("Creating a patch with merge patchType")
+				err = hiveClient.Create(ctx, SyncSetPatch)
+				Ω(err).ShouldNot(HaveOccurred())
+
+				err = waitForSyncSetInstanceApplied("test-syncpatch", "syncset", clusterNamespace)
+				Ω(err).ShouldNot(HaveOccurred())
+
+				By("Verifying the resource was patched")
+				resultConfigMap = &corev1.ConfigMap{}
+				err = targetClusterClient.Get(ctx, client.ObjectKey{Name: "foo", Namespace: "default"}, resultConfigMap)
+				Ω(err).ShouldNot(HaveOccurred())
+				Ω(resultConfigMap.Data).Should(Equal(map[string]string{"foo": "new-bar"}))
+
+				By("Modifying the patch type to strategic")
+				patch := `{ "data": { "foo": "baz-strategic" } }`
+				patchType := "strategic"
+				modifySynSetPatchType(hiveClient, SyncSetPatch, patch, patchType)
+
+				err = waitForSyncSetInstanceApplied("test-syncpatch", "syncset", clusterNamespace)
```

Comment:
This wait may not do what you expect. The syncset was already applied before you modified it. So `.Status.Applied` is already true even before your latest change has synced. To truly check whether your specific change was applied, you would have to ensure that the syncsetinstance status contains the hash of the latest patch content. It may be easier to just create a new syncset for each type of patch (after deleting the previous one).